### ISSUE TITLE
Tag update to ExpressRoute Gateway causes all ER circuits to become disconnected

### ISSUE THREAD

**Issue by robertdias** [hashicorp/terraform-provider-azurerm#13368]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

2.73.0

### Affected Resource(s)

* `azurerm_express_route_connection`
* `azurerm_express_route_gateway`

### Terraform Configuration Files

Imported from workflow:
```yml
           TF_VAR_azure_tags : "{
             Environment  = \"Production\",
             Repo         = \"infra\"
             }"
```
tfvars map:
```hcl
er_gateway_list = {
  "P-TX-DALLAS-1A" = {
    resource_group_name = "P-TX-NET-WAN"
    location            = "usgovtexas"
    virtual_hub_name    = "P-TX-HUB-1A"
    scale_units         = 5
  },
```
main.tf
```hcl
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "=2.73.0"
    }
  }
}

terraform {
  backend "azurerm" {}
}

provider "azurerm" {
  skip_provider_registration = "true"
  features {}
}

...

resource "azurerm_express_route_gateway" "er_gateways" {
  for_each = var.er_gateway_list

  name                = each.key
  resource_group_name = each.value.resource_group_name
  location            = each.value.location
  virtual_hub_id      = azurerm_virtual_hub.virtual_hub[each.value.virtual_hub_name].id
  scale_units         = each.value.scale_units
  tags                = var.azure_tags
}
```
plan and apply:
```hcl
Initializing the backend...

Successfully configured the backend "azurerm"! Terraform will automatically
use this backend unless the backend configuration changes.

Initializing provider plugins...
- Finding hashicorp/azurerm versions matching "2.73.0"...
- Finding hashicorp/azuread versions matching "1.6.0"...
- Installing hashicorp/azurerm v2.73.0...
- Installed hashicorp/azurerm v2.73.0 (signed by HashiCorp)
- Installing hashicorp/azuread v1.6.0...
- Installed hashicorp/azuread v1.6.0 (signed by HashiCorp)

Terraform has created a lock file .terraform.lock.hcl to record the provider selections 
it made above. Include this file in your version control repository so that Terraform can
 guarantee to make the same selections by default when you run 
"terraform init" in the future.

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" 
to see any changes that are required for your infrastructure. All Terraform commands 
should now work.

If you ever set or change modules or backend configuration for Terraform, rerun this
 command to reinitialize your working directory. If you forget, other commands will 
detect it and remind you to do so if necessary.

...
azurerm_express_route_gateway.er_gateways["P-TX-DALLAS-1A"]: Refreshing state... [id=/subscriptions/***/resourceGroups/P-TX-NET-WAN/providers/Microsoft.Network/expressRouteGateways/P-TX-DALLAS-1A]
...


Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  ~ update in-place
-/+ destroy and then create replacement
 <= read (data resources)

Terraform will perform the following actions:

  # azurerm_express_route_gateway.er_gateways["P-TX-DALLAS-1A"] will be updated in-place
  ~ resource "azurerm_express_route_gateway" "er_gateways" ***
        id                  = "/subscriptions/***/resourceGroups/P-TX-NET-WAN/providers/Microsoft.Network/expressRouteGateways/P-TX-DALLAS-1A"
        name                = "P-TX-DALLAS-1A"
      ~ tags                = ***
          + "Repo"        = "infra"
            # (1 unchanged element hidden)
        ***
        # (4 unchanged attributes hidden)
    ***
...
...
azurerm_express_route_gateway.er_gateways["P-TX-DALLAS-1A"]: Modifications complete after 15m22s [id=/subscriptions/***/resourceGroups/P-TX-NET-WAN/providers/Microsoft.Network/expressRouteGateways/P-TX-DALLAS-1A]
```

### Expected Behavior

We expected the ER Gateway to have the 'infra' tag applied on it. 'Update-in-place'

### Actual Behavior

The tag was applied, then all existing ER Connections were disconnected. This caused a complete outage for customer connectivity into Azure.

### Steps to Reproduce

1. Apply a tag to an existing ER gateway which has ER Connections into it.

### Important Factoids
- This is running in Azure Government.
- The ExpressRoute connections are maintained in a separate repo and tfstate file.

### References

* #0000

**Comment by seanlok** [comment]
This Happen when the scale unit is updated from code too.

**Comment by xuzhang3** [comment]
@paul-hugill  connection lost is because the GW updating operation ignored connections when send request body, result in service, the update operation will remove all the connections. But we cannot just set the connections back to the request body as SDK have implement the `MarshalJSON()` for  `ExpressRouteGatewayProperties` which will ignore the connections in the request body, this issue should fix the SDK issue first. You can find the `ExpressRouteGatewayProperties.MarshalJSON()` at  line 15296 in https://github.com/hashicorp/terraform-provider-azurerm/blob/main/vendor/github.com/Azure/azure-sdk-for-go/services/network/mgmt/2021-02-01/network/models.go.

**Comment by hriaz** [comment]
I can confirm this happens under Azure Virtual WAN as well for Express Route circuits. A tag update caused 3 of the customer's production Express Routes down. Considering the operation is a long-running operation, it took almost 24 minutes to re-connect everything. Not fun in global infrastructure.

**Comment by bubbletroubles** [comment]
Is this issue resolved now? AzureRM Provider [3.5.0](https://github.com/hashicorp/terraform-provider-azurerm/releases/tag/v3.5.0) has included the updated SDK which was required to fix this.

**Comment by xuzhang3** [comment]
@bubbletroubles I submit a PR #16680   to fix this issue.
